fix watching external changes on other than input element, reenable/improve some unit tests

kferenc617 opened the pull request:
Added getGetterSetter to AutoNumericHelper.js and call it from the constructor for the 'watch external changes' functionality

After this fix, the test suite 'Managing external changes' could be reenabled and it works perfectly.

I've also fixed one spec related to event bubbling (minor fix was needed) and implemented a test case according to the TODO comment related to decimalPlacesShownOnBlur functionality.

Please let me know if I could help merging my PRs.

kferenc617 committed:
fix watching external changes on other than input element, reenable/improve some unit tests

AlexandreBonneau commented on test/unit/autoNumeric.spec.js:
```diff
@@ -1573,9 +1573,12 @@ describe('autoNumeric options and `options.*` methods', () => {
         });
 
-        xit('should not modify how non-AutoNumeric or input event bubble up', () => { //FIXME The `testingFunc()` is called, but somehow Jasmine spy does not pick that up
+        it('should not modify how non-AutoNumeric or input event bubble up', () => { //FIXME The `testingFunc()` is called, but somehow Jasmine spy does not pick that up
```
Please remove the comment here, since it's not applicable now

kferenc617 replied:
Removed the unnecessary comment

AlexandreBonneau commented on test/unit/autoNumeric.spec.js:
```diff
@@ -1562,9 +1562,9 @@ describe('autoNumeric options and `options.*` methods', () => {
-            input.addEventListener('click', customFunction.testingFunc);
+            // Event listeners should be added after customFunction.testingFunc has been 'decorated' by the spyOn API
+            // form.addEventListener('click', customFunction.testingFunc);
+            // input.addEventListener('click', customFunction.testingFunc);
```
Please remove the comments

kferenc617 replied:
Also removed these unnecessary comment

AlexandreBonneau commented on src/AutoNumeric.js:
```diff
@@ -209,7 +209,7 @@ export default class AutoNumeric {
         this.internalModification = false; // This is temporarily set to `true` only when the AutoNumeric object does update the element value
         this.attributeToWatch = this._getAttributeToWatch();
-        this.getterSetter = Object.getOwnPropertyDescriptor(this.domElement.__proto__, this.attributeToWatch);
+        this.getterSetter = AutoNumericHelper.getGetterSetter(this.domElement, this.attributeToWatch);  // Object.getOwnPropertyDescriptor(this.domElement.__proto__, this.attributeToWatch);
```
Please remove the comment

kferenc617 replied:
And removed this one :)

AlexandreBonneau reviewed (changes requested):
Thanks again for the PR

kferenc617 committed:
Removed unnecessary comments

AlexandreBonneau committed:
Add docs to the new getGetterSetter function

Adds docs to the new getGetterSetter function

Updates the spacing to conform to the code style
